db-engine/src/main/java/com/splicemachine/db/impl/sql/conn/GenericLanguageConnectionContext.java

```diff
@@ -1333,6 +1333,10 @@ public void removeStatement(GenericStatement statement) throws StandardException
     public PreparedStatement lookupStatement(GenericStatement statement) throws StandardException{
         GenericStorablePreparedStatement ps = getDataDictionary().getDataDictionaryCache().statementCacheFind(statement);
         if (ps==null) {
+            String sqlText = statement.getSource().toUpperCase();
+            if ((sqlText.startsWith("CREATE") || sqlText.startsWith("DECLARE")) && sqlText.contains("IF NOT EXISTS")){
+                return null;
+            }
```

Comment:
Why do we need this?

Reply:
CREATE TABLE statements are cached if they have run through once. So if we run `CREATE TABLE IF NOT EXISTS T ...` twice, the second execution will bypass all checks in bind phase, including the newly added "if not exists" check. As a result, the execution returns an error. That should not be the case when "if not exists" presents.

It seems that CREATE SCHEMA statement is never cached, that's why "if not exists" works just fine there.

An alternative would be to pass this createBehavior all the way down to the engine and check there. Not sure that's a better way, though.

Reply:
I just found that it's better to remove the hack in statement cache and add the check in its constant action. Will improve it with a new commit.